index.js

```diff
@@ -142,22 +142,39 @@ function createMainWindow() {
 	win.on("closed", onClosed);
 
 	win.on("move", () => {
+		if (win.isMaximized()) return;
 		let position = win.getPosition();
-		config.set("window-position", { x: position[0], y: position[1] });
+		lateSave("window-position", { x: position[0], y: position[1] });
 	});
 
+	let winWasMaximized;
+
 	win.on("resize", () => {
 		const windowSize = win.getSize();
 
-		config.set("window-maximized", win.isMaximized());
-		if (!win.isMaximized()) {
-			config.set("window-size", {
+		const isMaximized = win.isMaximized();
+		if (winWasMaximized !== isMaximized) {
+			winWasMaximized = isMaximized;
+			config.set("window-maximized", isMaximized);
+		}
+		if (!isMaximized) {
+			lateSave("window-size", {
 				width: windowSize[0],
 				height: windowSize[1],
 			});
 		}
 	});
 
+	let savedTimeouts = {};
+	function lateSave(key, value) {
```

Comment:
nit: the debouncing can be improved by applying immediately the save then waiting before another save occurs:
```suggestion
function debouncedSave(key, value, timeout = 1000){
    if (!savedTimeouts[key]) {
      config.set(key, value);
    }
    clearTimeout(savedTimeouts[key]);
    delete savedTimeouts[key];
    …
  };
}
```

Reply:
I don't understand how would that work?

how would the function know if it should immediately save or timeout save?

from what I understand from your proposition, it would always immediately saves which defeats the purpose of this

(also tbh I don't think its really useful to immediately save instead of waiting 1 second and then saving - if you think the delay is too much it could just be lowered)

Reply:
The idea of debouncing is to save immediately if no save has been performed in the last second, otherwise wait (based on the timeout being set or not)  - here, unless I missed something, waiting each time has edge cases - eg if you quit the app just after having resized it, it won't be saved because the timeout did not finish.

Reply:
I'm not sure  you understand the way the events are fired

If a user resize his window by 50px - it will send 50 move events.
if you immediately save the first value you get it will just be currentValue + 1px (useless)
and the last event will **not** be saved immediately (because there were 50 events before it) - so that edge case you talked about will still apply..

Anyways I don't think that edge case is very realistic, its pretty hard to resize/move the window and then close it in less than one second (also its a weird thing to do, even if you could)

also like I said above, the delay could be lowered to something like 750milisecs so that there is an even lesser chance of that edge case happing